src/providers/pkcs11_provider/key_management.rs

```diff
@@ -62,6 +62,53 @@ impl Pkcs11Provider {
             }
         }
     }
+
+    pub(super) fn move_pub_key_to_psa_crypto(&self, key_triple: &KeyTriple) -> Result<Id> {
+        info!("Attempting to export public key");
+        let export_operation = psa_export_public_key::Operation {
+            key_name: key_triple.key_name().to_owned(),
+        };
+        let psa_export_public_key::Result { data } =
+            self.psa_export_public_key_internal(key_triple.app_name().clone(), export_operation)?;
+
+        info!("Importing public key into PSA Crypto");
+        let (_, mut attributes) = self.get_key_info(key_triple)?;
+        attributes.lifetime = Lifetime::Volatile;
+        attributes.key_type = match attributes.key_type {
+            Type::RsaKeyPair | Type::RsaPublicKey => Type::RsaPublicKey,
+            Type::EccKeyPair { curve_family } | Type::EccPublicKey { curve_family } => {
+                Type::EccPublicKey { curve_family }
+            }
+            Type::DhKeyPair { group_family } | Type::DhPublicKey { group_family } => {
+                Type::DhPublicKey { group_family }
+            }
+            _ => return Err(ResponseStatus::PsaErrorInvalidArgument),
+        };
+        let id = psa_crypto::operations::key_management::import(attributes, None, &data)?;
+
+        Ok(id)
+    }
+
+    pub(super) fn remove_psa_crypto_pub_key(
+        &self,
+        key_triple: &KeyTriple,
+        pub_key_id: Id,
+    ) -> Result<()> {
+        info!("Removing public key stored in PSA.");
+        unsafe { psa_crypto::operations::key_management::destroy(pub_key_id) }.map_err(|e| {
+            if crate::utils::GlobalConfig::log_error_details() {
+                error!(
+                    "Failed to remove public part of key \"{}\" from PSA Crypto.",
+                    key_triple.key_name()
```

Comment:
I would personally say that if `key_triple` is passed as a parameter just for logging, we can just skip that and log in all cases: `Failed to remove public key from PSA Crypto.`
But feel free to keep it if you think this info is definitely needed.

Reply:
Yeah, I was a bit 😒 changing that just to print the key triple, but could change it back

Reply:
Changed!